app/scripts/views/mixins/settings-panel-mixin.js

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// helper functions for views with a profile image. Meant to be mixed into views.
+
+define([
+  'jquery',
+  'views/base'
+], function ($, BaseView) {
+  'use strict';
+
+  return {
+    initialize: function (options) {
+      this.superView = options.superView;
+    },
+    events: {
+      'click .settings-unit-toggle': BaseView.preventDefaultThen('_triggerPanel'),
+      'click .cancel': BaseView.preventDefaultThen('closePanelReturnToSettings')
+    },
+
+    _triggerPanel: function (event) {
+      var href = event && $(event.currentTarget).data('href');
+      if (href) {
+        this.navigate(href);
+      }
+    },
+
+    openPanel: function () {
+      var unit = this.$('.settings-unit');
+      unit.addClass('open');
+    },
+
+    closePanelReturnToSettings: function () {
+      this.navigate('settings');
+      this.closePanel();
+    },
+
+    closePanel: function () {
+      this.$('.settings-unit').removeClass('open');
+    },
+
+    displaySuccess: function (msg) {
+      var self = this;
+      if (! self.superView) {
+        return;
+      }
+      self.superView.displaySuccess(msg);
+      self.closePanel();
```

Comment:
> Seems like that would be leaking implementation details in to the view,

I disagree on this point. Closing the panel on `displaySuccess` is a pretty serious side effect, something a view implementer has to keep track of: "if I call displaySuccess here, what's going to happen?" I personally wouldn't expect the panel to close on me automatically. I'd expect the success message to be displayed, in the panel, similarly to non-panel views.

It seems like this really belongs as a new function, and displaySuccess should be left alone so panels can display success or error messages, in content, if needed.

`displaySuccesInParent` or `closeAndDisplaySuccess`?